modules/porous_flow/src/bcs/PorousFlowEnthalpySink.C

```diff
@@ -0,0 +1,117 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowEnthalpySink.h"
+#include "Function.h"
+#include "PorousFlowDictator.h"
+#include "SinglePhaseFluidProperties.h"
+#include "MooseVariable.h"
+#include "PorousFlowSinkBC.h"
+
+#include "libmesh/quadrature.h"
+
+#include <iostream>
+
+registerMooseObject("PorousFlowApp", PorousFlowEnthalpySink);
+
+InputParameters
+PorousFlowEnthalpySink::validParams()
+{
+  InputParameters params = IntegratedBC::validParams();
+  params += PorousFlowSinkBC::validParamsCommon();
+  params.addRequiredParam<UserObjectName>(
+      "PorousFlowDictator", "The UserObject that holds the list of PorousFlow variable names");
+  params.addParam<unsigned int>("fluid_phase",
+                                "Evaluate enthalpy at the pressure of this fluid phase.");
+  params.addClassDescription(
+      "Applies a source equal to the product of the mass flux and the "
+      "fluid enthalpy. The enthalpy is computed at temperature T_in and pressure equal to the "
+      "porepressure in the porous medium. Hence this adds heat energy "
```

Comment:
"porepressure in the porous medium, if fluid_phase is given, otherwise at the supplied porepressure.  Hence this adds heat energy "

Reply:
✅